src/ann_solo/decoy_generator.py

```diff
@@ -0,0 +1,186 @@
+
+
+def shuffle_and_reposition(spectrum: MsmsSpectrum) -> MsmsSpectrum:
+    """
+    Creates a decoy spectrum from a real spectrum.
+
+    Parameters
+    ----------
+    spectrum: MsmsSpectrum
+        Real spectrum.
+    Returns
+    -------
+    MsmsSpectrum
+        Decoy spectrum.
+    """
+    # annotate original spectrum
+    spectrum.annotate_proforma(spectrum.peptide, config.fragment_tol_mass,
+                               config.fragment_tol_mode, "aby",
+                               neutral_losses=True)
+    # parse original spectrum
+    parsed_sequence = proforma.parse(spectrum.proforma)
+    shuffled_sequence, shuffled_sequence_mapping = _shuffle(
+        parsed_sequence[0].sequence)
+
+    # Compute theoretical fragment m/z of the original peptide
+    genuine_peptide_theoretical_fragments = {
+    str(ion.ion_type) + '^' + str(ion.charge): mz for ion, mz in
+    fragment_annotation.get_theoretical_fragments(
+        parsed_sequence[0], ion_types="aby",
+        max_charge=spectrum.precursor_charge,
+        neutral_losses=fragment_annotation._neutral_loss)}
+
+    # Constract decoy proteoform
+    if parsed_sequence[0].modifications is not None:
+        for modification in parsed_sequence[0].modifications:
+            setattr(modification, 'position',
+                    shuffled_sequence_mapping[modification.position])
+    decoy_proforma = proforma.Proteoform(shuffled_sequence,
+                                         parsed_sequence[0].modifications)
+
+    # Compute theoretical fragment m/z of the shuffled peptide
+    decoy_theoretical_fragments = {
+    str(ion.ion_type) + '^' + str(ion.charge): mz for ion, mz in
+    fragment_annotation.get_theoretical_fragments(
+        decoy_proforma, ion_types="aby",
+        max_charge=spectrum.precursor_charge,
+        neutral_losses=fragment_annotation._neutral_loss)}
+
+    # Initialize decoy fragment arrays
+    mz_shuffled, intensity_shuffled, annotation_shuffled = np.zeros_like(
+        spectrum.mz), np.zeros_like(spectrum.intensity), np.full_like(
+        spectrum.annotation, None, object)
+    # Initialize new original fragment annotation array
+    annotation_original = np.full_like(spectrum.annotation, None, object)
+    for i in range(len(spectrum.mz)):
+        peak_annotation = str(spectrum.annotation[i][0].ion_type) + '^' + str(
+            spectrum.annotation[i][0].charge)
+        # Assign fragment values
+        intensity_shuffled[i] = spectrum.intensity[i]
+        mz_shuffled[i] = spectrum.mz[i]
+        annotation_shuffled[i] = None if str(spectrum.annotation[i]) == '?' \
+                                else fragment_annotation.FragmentAnnotation(
+                                spectrum.annotation[i][0].ion_type,
+                                charge=spectrum.annotation[i][0].charge)
+
+        # To maintain FragmentAnnotation obj for all annotations
+        annotation_original[i] = None if str(spectrum.annotation[i]) == '?' \

```

Comment:
**question:** I don't immediately understand why the peak annotations for the original spectrum would change?